src/m365/spo/commands/homesite/homesite-set.ts

```diff
@@ -77,26 +70,151 @@ class SpoHomeSiteSetCommand extends SpoCommand {
           'content-Type': 'application/json'
         },
         responseType: 'json',
-        data: {
-          sphSiteUrl: args.options.siteUrl
-        }
+        data: {}
       };
 
-      if (args.options.vivaConnectionsDefaultStart !== undefined) {
-        requestOptions.url += '/SetSPHSiteWithConfiguration';
-        requestOptions.data.configuration = { vivaConnectionsDefaultStart: args.options.vivaConnectionsDefaultStart };
+      const homeSiteCount = await this.getHomeSiteCount(spoAdminUrl, logger);
+
+      if (args.options.vivaConnectionsDefaultStart !== undefined ||
+        args.options.draftMode !== undefined ||
+        args.options.audienceIds !== undefined ||
+        args.options.audienceNames !== undefined ||
+        args.options.targetedLicenseType !== undefined ||
+        args.options.order !== undefined) {
```

Comment:
I think this is not the way we should implement it. Currently the legacy code and the new code are all mixed together which will make it quite hard to remove it afterwards.

I think the goal should be that we create 2 clear forks in the code. One with legacy code that only runs when a few conditions are met, and one branch with the new code and new API.

Reply:
How do you propose to do it or shall we wait until September when we can remove the old code to make it cleaner and not duplicating efforts?

Reply:
If you feel more comfortable removing the legacy code right away and implementing the breaking changes, I'm fine with that. In that case, I will mark this issue as a v11 issue. Your call 😊
If we choose this approach, we'll have to update the related issue as well, but that's no problem for me.

Reply:
Thanks so much for the review, the two branches have to coexist until September which is next month only. Happy to remove the legacy code to have a cleaner code and marked for next release.